plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+        self.log('Access Point Configuration Information: ' + str(ap_config_data), "INFO")
+        if self.payload["ap_selected_field"] == "" or self.payload["ap_selected_field"] == "all":
+            return ap_config_data
+
+        fields = [str(self.keymap[x]) for x in self.payload["ap_selected_field"].split(",")]
+        if len(ap_config_data) != 0:
+            df = pd.DataFrame.from_records(ap_config_data)
+            selected_data = df[fields]
+            return selected_data.to_dict('records')
+        else:
+            return None
+
+    def compare_ap_cofig_with_inputdata(self, apconfig):
+        """
+        This function used to compare with the input ap detail with the current ap configuration
+        information are not same, those data will be updated in the AP input information.
+        Parameters:
+          - apconfig: This is response of the get_ap_configuration
+        Returns:
+            This will be the return the final data for update AP detail.
+            "final_input": [
+                {
+                    "adminStatus": true,
+                    "apList": [
+                        {
+                            "macAddress": "34:5d:a8:0e:20:b4"
+                        }
+                    ],
+                    "configureLedBrightnessLevel": true,
+                    "ledBrightnessLevel": 4,
+                    "macAddress": "34:5d:a8:0e:20:b4"
+                } ]
+        Example:
+            functions = Accesspoint(module)
+            final_input_data = functions.compare_ap_cofig_with_inputdata(all_apconfig)
+        """
+        final_apchange = []
+        for each_input in self.payload["config"]:
+            for eachap in apconfig:
+                # We are identifing AP based on the AP mac Address so we cannot update this field.
+                if each_input["mac_address"] == eachap["macAddress"]:
+                    newdict = {}
+                    for each_key in list(each_input.keys()):
+                        if each_input[each_key] != eachap[self.keymap[each_key]]:
+                            if each_key == "ap_name":
+                                newdict[self.keymap[each_key]] = eachap[self.keymap[each_key]]
+                                newdict[self.keymap[each_key] + "New"] = each_input[each_key]
+                            else:
+                                newdict[self.keymap[each_key]] = each_input[each_key]
+
+                    if newdict:
+                        newdict["macAddress"] = eachap["ethMac"]
+                        final_apchange.append(newdict)
+        if len(final_apchange) > 0:
+            return final_apchange
+        else:
+            self.log('Input Access Point Configuration remain same in Current AP configration',
+                      "INFO")
+            return None
+
+    def get_want(self, device_data):
+        update_apconfig = self.update_ap_configuration(device_data)
```

Comment:
This function we used to collect the input parameter given in the playbook.

Reply:
However we need to get the device details based on the input file and validation new key value only update after update.